src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/DuplicateTypeNameAnalyzer.cs

```diff
@@ -148,11 +148,33 @@ public override void Analyze(ISymbolAnalysisContext context)
                 var currentFullTypeName = namedTypeSymbol.ToDisplayString();
                 var currentAssemblyName = namedTypeSymbol.ContainingAssembly?.Name ?? "";
                 
-                // Don't report if it's exactly the same type
-                if (string.Equals(currentFullTypeName, qualifiedTypeName, StringComparison.Ordinal) && 
-                    string.Equals(currentAssemblyName, packageName, StringComparison.Ordinal))
+                // Only flag types that have a non-generic name match with the reserved type
```

Comment:
but we also want to detect conflicts in generic names. Why don't you simply compare the names. Forgot about the generic/non-generic change. Generic names are simply suffixed with apostrophe followed with the number of generic parameters